vision/localize_objects.rb

```diff
@@ -0,0 +1,91 @@
+# Copyright 2018 Google, Inc
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+def localize_objects image_path:
+  # [START vision_localize_objects]
+  # image_path = "Path to local image file, eg. './image.png'"
+
+  require "google/cloud/vision"
+
+  vision = Google::Cloud::Vision::V1.new
+
+  image_content = IO.read image_path, mode: 'rb'
+  image = Google::Cloud::Vision::V1::Image.new content: image_content
+
+  feature = Google::Cloud::Vision::V1::Feature.new type: :OBJECT_LOCALIZATION
+
+  request = Google::Cloud::Vision::V1::AnnotateImageRequest.new image: image, features: [feature]
+
+  response = (vision.batch_annotate_images [request]).responses.first
```

Comment:
Let's separate the `.responses.first` out explicitly

`batch_annotate_images` isn't necessarily intuitive

Reply:
tl;dr The whole thing should come together looking something like:

```ruby
# image_path = "Path to local image file, eg. './image.png'"

require "google/cloud/vision"

vision = Google::Cloud::Vision.new

request = {
  features: [:OBJECT_LOCALIZATION],
  image: {
    content: File.binread(image_file)
  }
}

result  = vision.batch_attotate_images [request]
response = result.responses.first
objects  = response.localized_object_annotations

object.each do |onject|
# ...
```

Reply:
done